scipy/integrate/tests/test_quadrature.py

```diff
@@ -524,6 +527,324 @@ def func(x):
+                -1,
+                np.array(
+                    [
+                        [1 / 3, 8 / 3, 9.0, 64 / 3],
+                        [2 / 3, 3.0, 28 / 3, 65 / 3],
+                        [1.0, 10 / 3, 29 / 3, 22.0],
+                    ]
+                ),
+            ),
+            (
+                # Multi dimensional input for dx
+                np.array([[1, 4, 9, 16], [1, 4, 9, 16], [1, 4, 9, 16]]),
+                [[1.0], [2.0], [1.0]],
+                None,
+                None,
+                np.array(
+                    [
+                        [7 / 3, 26 / 3, 21.0],
+                        [14 / 3, 52 / 3, 42.0],
+                        [7 / 3, 26 / 3, 21.0],
+                    ]
+                ),
+            ),
+            (
+                # Multi dimensional values for initial and dx with non-default
+                # axis, and mixture of int and float inputs
+                np.array([[1, 4, 9, 16], [1, 4, 9, 16], [1, 4, 9, 16]]),
+                [[1.0, 2.0, 1.0, 5.0]],
+                [[0, 0, 0, 1]],
+                0,
+                np.array(
+                    [
+                        [0, 0, 0, 1],
+                        [1, 8, 9, 81],
+                        [2, 16, 18, 161],
+                    ]
+                ),
+            ),
+            (
+                # Test case with cubic y
+                np.array([[0, 1, 8]]),
+                1,
+                None,
+                None,
+                np.array([[0.0, 4.0]]),
+            ),
+        ],
+    )
+    def test_simpson_multi_dx(self, y, dx, initial, axis, int_y):
+        kwargs = {}
+        if dx is not None:
+            kwargs["dx"] = dx
+        if initial is not None:
+            kwargs["initial"] = initial
+        if axis is not None:
+            kwargs["axis"] = axis
+        assert_allclose(
+            cumulative_simpson(y, **kwargs),
+            int_y,
+        )
+
+    @pytest.mark.parametrize(('message', 'kwarg_update'), [
+        ("x must be monotonically increasing", dict(x=[2, 2, 3, 4])),
+        ("x must be monotonically", dict(x=[x0, [2, 2, 4, 8]], y=[y0, y0])),
+        ("x must be monotonically", dict(x=[x0, x0, x0], y=[y0, y0, y0], axis=0)),
+        ("At least 3 points are required", dict(x=x0[:2], y=y0[:2])),
+        ("`axis=4` is not valid for `y` with `y.ndim=1`", dict(axis=4)),
+        ("shape of `x` must be the same as `y` or 1-D", dict(x=np.arange(5))),
+        ("`initial` must either be a scalar or...", dict(initial=np.arange(5))),
+        ("`dx` must either be a scalar or...", dict(x=None, dx=np.arange(5))),
+    ])
+    def test_simpson_exceptions(self, message, kwarg_update):
+        kwargs0 = dict(y=self.y0, x=self.x0, dx=None, initial=None, axis=-1)
+        with pytest.raises(ValueError, match=message):
+            cumulative_simpson(**dict(kwargs0, **kwarg_update))
+
+    def _get_theoretical_diff_between_simps_and_cum_simps(self, y, x):
```

Comment:
@mdhaber , 
I want to update this function by importing and using integrate._cumulative_simpson_unequal_intervals instead of adding the maths here.
But the function is not in the __all__ list of _quadrature.py so it does not get imported. It also should not be since it should not be accessible by users.
I can get around this by moving _cumulative_simpson_unequal_intervals  to a separate file.
Is there another way around this?

Reply:
You can add `from scipy.integrate._quadrature import _cumulatively_sum_simpson_integrals` at the top of the file.